### DISCUSSION TITLE
M4 Mac mini nested virtualization performance benchmark

### DISCUSSION THREAD

**Discussion by GainSury** [utmapp/UTM#6860] [Show and tell]
## Test Env
 - EL0: MacOS 15.1 Mac mini 16GB 256GB
 - EL1: UTM 4.6 beta + Fedora Server aarch64 41-1.4 10GB
 - EL2: QEMU + KVM Fedora Server aarch64 41-1.4 8GB
 - M2 Mac Mini
 - M1 Macbook Pro 
## BenchMark:
 - Geekbench 6
 - fio

## fio args
fio --name=test --directory=/path/to/dir --rw=randrw --bs=4k --size=1G --numjobs=1 --time_based —runtime=30 --group_reporting
(m1 macbook pro with —runtime=90)

## Results
| Device                | Read IOPS | Read Speed (MB/s) | Write IOPS | Write Speed (MB/s) | Geekbench6 ST |
|-----------------------|-----------|-------------------|------------|--------------------|---------------|
| M4 Mac mini          | 8647      | 35.4              | 8660       | 35.5               | 3925          |
| UTM Appl Virt (EL1)  | 3532      | 14.5              | 3526       | 14.4               | 3362          |
| QEMU/KVM (nested/EL2)| 2606      | 10.7              | 2593       | 10.6               | 2875          |
| M1 MacBook Pro       | 10400     | 40.7              | 10400      | 40.7               | 2402          |
| M2 Mac mini          | 5227      | 21.4              | 5230       | 21.4               | 2608          |
| **EL0 vs EL1**       | **40.85%**| **40.96%**        | **40.72%** | **40.56%**         | **85.66%**    |
| **EL0 vs EL2**       | **30.14%**| **30.23%**        | **29.94%** | **29.86%**         | **73.25%**    |
| **EL1 vs EL2**       | **73.78%**| **73.79%**        | **73.54%** | **73.61%**         | **85.51%**    |

**Comment by Lewiscowles1986** [comment]
Anyone got the M4 (mac air or pro) and ran this again?
